tests/core/test_variable.py

```diff
@@ -0,0 +1,558 @@
+        self.assertDictEqual({p:v for p,v in w.depends}, {'t':t,'u':u,'v':v})
+        self.assertAlmostEqual(w.value, 6.0)
+
+    def test_dependency_graph(self):
+        """Test dependency_graph method."""
+        t = relentless.DesignVariable(value=1.0)
+        u = relentless.DesignVariable(value=2.0)
+        v = relentless.DesignVariable(value=3.0)
+
+        #test 1 level of dependency
+        w = DepVar(t=t, u=u, v=v)
+        g = w.dependency_graph()
+        self.assertCountEqual(g.nodes, [t,u,v,w])
+        self.assertCountEqual(g.edges, [(w,t),(w,u),(w,v)])
+        self.assertCountEqual([g.edges[a,b]['params'] for (a,b) in g.edges],
+                              [['t'],['u'],['v']])
+
+        #test 2 levels of dependency
+        x = DepVar(t=t, v=v)
+        y = DepVar(w=w, x=x)
+        g = y.dependency_graph()
+        self.assertCountEqual(g.nodes, [t,u,v,w,x,y])
+        self.assertCountEqual(g.edges, [(w,t),(w,u),(w,v),(x,t),(x,v),(y,w),(y,x)])
+        self.assertCountEqual([g.edges[a,b]['params'] for (a,b) in g.edges],
+                              [['t'],['u'],['v'],['t'],['v'],['w'],['x']])
+
+        #test more complex dependency
+        z = DepVar(t=t, w=w, y=y)
+        g = z.dependency_graph()
+        self.assertCountEqual(g.nodes, [t,u,v,w,x,y,z])
+        self.assertCountEqual(g.edges, [(w,t),(w,u),(w,v),(x,t),(x,v),(y,w),(y,x),
+                                        (z,t),(z,w),(z,y)])
+        self.assertCountEqual([g.edges[a,b]['params'] for (a,b) in g.edges],
+                              [['t'],['u'],['v'],['t'],['v'],['w'],['x'],['t'],['w'],['y']])
+
+        #test 'multiple' dependency on same object
+        q = DepVar(t=t, u=t, v=t)
+        g = q.dependency_graph()
+        self.assertCountEqual(g.nodes, [t,q])
+        self.assertCountEqual(g.edges, [(q,t)])
+        params = [g.edges[a,b]['params'] for (a,b) in g.edges][0]
+        self.assertCountEqual(params, ['t','u','v'])
+
+        q = DepVar(t=t, u=t, v=v, w=v)
+        g = q.dependency_graph()
+        self.assertCountEqual(g.nodes, [t,v,q])
+        self.assertCountEqual(g.edges, [(q,t),(q,v)])
+        params0 = sorted([g.edges[a,b]['params'] for (a,b) in g.edges][0])
+        params1 = sorted([g.edges[a,b]['params'] for (a,b) in g.edges][1])
+        self.assertCountEqual([params0,params1], [['t','u'],['v','w']])
+
+        #test circular dependencies
+        a = DepVar(t=t)
+        a.t = DepVar(t=a)
+        g = a.dependency_graph()
+        self.assertCountEqual(g.nodes, [a.t,a])
+        self.assertCountEqual(g.edges, [(a,a.t),(a.t,a)])
+        self.assertCountEqual([g.edges[a,b]['params'] for (a,b) in g.edges],
+                              [['t'],['t']])
+
+        a = DepVar(t=t)
+        b = DepVar(t=a)
+        c = DepVar(t=b)
+        a.t = DepVar(t=c)
+        g = a.dependency_graph()
+        self.assertCountEqual(g.nodes, [a.t,a,b,c])
+        self.assertCountEqual(g.edges, [(b,a),(c,b),(a.t,c),(a,a.t)])
```

Comment:
You can probably make this just a 3-ring cycle a, b, c (not a 4 ring cycle) then set `a.t = c`. The checks would then be:
```suggestion
        self.assertCountEqual(g.nodes, [a,b,c])
        self.assertCountEqual(g.edges, [(b,a),(c,b),(a,c)])
```
You can probably remove the `params` check, since it isn't testing anything new.